### PR TITLE
flathub integration request - Nitrokey App

### PR DESCRIPTION
**Pull request by daringer**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. 
`device=all` was included in order to support the wide range of security-keys we are supporting now and in the future
- [x] All assets referenced in the manifest are redistributable by any party. 
- [x] I am an upstream contributor to the project. 
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] No additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.nitrokey.nitrokey-app

**Comment by daringer** [comment]
sorry my fault, did not run validate, now providing appdata through this repository, will be updated inside the application on next release, validate now also working:
```
$ flatpak run org.freedesktop.appstream-glib validate  build-dir/files/share/appdata/com.nitrokey.nitrokey-app.appdata.xml
build-dir/files/share/appdata/com.nitrokey.nitrokey-app.appdata.xml: OK
```

**Comment by barthalion** [comment]
bot, build com.nitrokey.nitrokey-app

File: README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# com.nitrokey.nitrokey-app - Nitrokey App Flatpak
```
Comment (barthalion) [inline README.md:1]: This does not seem needed here.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+local-install:
```
Comment (barthalion) [inline Makefile:1]: I'm usually against helper Makefiles in Flathub repos but it's up to you.

File: com.nitrokey.nitrokey-app.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  <content_rating type="oars-1.1"/>
+  <launchable type="desktop-id">nitrokey-app.desktop</launchable>
+  <url type="homepage">https://www.nitrokey.com/</url>
+  <url type="bugtracker">https://github.com/Nitrokey/nitrokey-app/issues</url>
+  <url type="faq">https://www.nitrokey.com/documentation/frequently-asked-questions</url>
+  <url type="help">https://www.nitrokey.com/start</url>
+  <screenshots>
+    <screenshot type="default">
+	    <image>https://raw.githubusercontent.com/Nitrokey/nitrokey-app/master/images/icon/nitrokey-app-icon-128-black.png</image>
+      <caption><!-- Describe this screenshot in less than ~10 words --></caption>
```
Comment (barthalion) [inline com.nitrokey.nitrokey-app.appdata.xml:38]:
```suggestion
```

File: com.nitrokey.nitrokey-app.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    buildsystem: simple
+    build-commands:
+      - "mv /app/share/icons/hicolor/scalable/apps/nitrokey-app.svg /app/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg"
+      - "mv /app/share/applications/nitrokey-app.desktop /app/share/applications/${FLATPAK_ID}.desktop"
+      - "desktop-file-edit --set-icon=${FLATPAK_ID} /app/share/applications/${FLATPAK_ID}.desktop"
+      - "mv ${FLATPAK_ID}.appdata.xml /app/share/metainfo/${FLATPAK_ID}.appdata.xml"
+    sources:
+      - type: file
+        path: com.nitrokey.nitrokey-app.appdata.xml
+        sha256: c7419a0cad827a1a5af82c8f0e6dfccfececa581fdb67a53bfefb0676b939f9d
```
Comment (barthalion) [inline com.nitrokey.nitrokey-app.yml:59]:
```suggestion
```

Local files don't need checksums, it's fine to drop it.

File: com.nitrokey.nitrokey-app.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+command: nitrokey-app
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --filesystem=host
+  - --device=all
+modules:
```
Comment (barthalion) [inline com.nitrokey.nitrokey-app.yml:11]: You should add a global `cleanup` array to remove unneeded files, something like this is a good starting point:

```
cleanup:
  - "*.a"
  - /include
```

File: com.nitrokey.nitrokey-app.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  - name: libusb
+    buildsystem: autotools
```
Comment (barthalion) [inline com.nitrokey.nitrokey-app.yml:13]: `autotools` is the default build system and doesn't need to be set explicitly.

File: flathub.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (barthalion) [inline flathub.json:2]: Any reason to limit it to x86_64?

Comment (daringer) [reply flathub.json:2]: as of now because it is not tested extensively

File: com.nitrokey.nitrokey-app.yml (hunk 6 above)
Comment (daringer) [reply com.nitrokey.nitrokey-app.yml]: done

File: com.nitrokey.nitrokey-app.yml (hunk 5 above)
Comment (daringer) [reply com.nitrokey.nitrokey-app.yml:17]: done and added more

File: com.nitrokey.nitrokey-app.yml (hunk 4 above)
Comment (daringer) [reply com.nitrokey.nitrokey-app.yml]: done

File: com.nitrokey.nitrokey-app.appdata.xml (hunk 3 above)
Comment (daringer) [reply com.nitrokey.nitrokey-app.appdata.xml]: updated with 2 screenshots + captions

File: Makefile (hunk 2 above)
Comment (daringer) [reply Makefile]: inherited from old non-flathub flatpak repo, personal anti-forget-mechanism, removed.

File: README.md (hunk 1 above)
Comment (daringer) [reply README.md]: done, removed

**Comment by daringer** [comment]
thx for the feedback, updated and pushed

**Comment by bilelmoussaoui** [comment]
bot, build com.nitrokey.nitrokey-app

**Comment by daringer** [comment]
Unfortunately there is no 64x64 icon included inside the latest release, pushed it into this repository. This will be fixed and then removed with our next release. Can I also start the build? bot, build com.nitrokey.nitrokey-app

**Comment by daringer** [comment]
struggling with icon paths, bot, build com.nitrokey.nitrokey-app

**Comment by daringer** [comment]
bot, build com.nitrokey.nitrokey-app

**Comment by daringer** [comment]
finally :smile:

**Pull request closed without merging**